lib/pf/lookup/person.pm

```diff
@@ -36,26 +39,31 @@ sub lookup_person {
     my $logger = get_logger();
     unless (defined $source_id) {
         $logger->info("undefined source id provided");
-        return "Unknown source";
+        return undef;
     }
     my $source = pf::authentication::getAuthenticationSource($source_id);
     if (!$source) {
        $logger->info("Unable to locate the source $source_id");
-       return "Unable to locate the source $source_id!\n";
+       return undef;
     }
 
     unless (person_exist($pid)) {
-        return "Person $pid is not a registered user!\n";
+        return undef;
     }
-    my ($person, $msg) = $source->search_attributes($pid);
-    if (!$person) {
-       $logger->info("Cannot search attributes for user '$pid'");
-       return "Cannot search attributes for user '$pid'!\n";
+    my $person = $CHI_CACHE->get("$source_id.$pid");
```

Comment:
If i use compute then each time the person has been found in the cache then i will do a person_modify and i don't want to do it.